Sync issues with and without price tags

xno-miner opened the pull request:
Resolves https://github.com/ubiquity/.github/issues/115

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

xno-miner committed:
Update github.ts

xno-miner committed:
Update index.ts

xno-miner committed:
Minor fix

xno-miner committed:
Update helpers/github.ts

Co-authored-by: アレクサンダー.eth <4975670+0x4007@users.noreply.github.com>

xno-miner committed:
Update devpool-issue-handler.test.ts

xno-miner committed:
Update devpool-issue-handler.test.ts

xno-miner committed:
Update index.ts

xno-miner committed:
Update github.ts

xno-miner committed:
Minor fix

xno-miner committed:
Minor clean

xno-miner committed:
Fix getting repo name

xno-miner committed:
Moved the DEVPOOL_ variables to .env (copy the values from .env.example)

xno-miner committed:
Moved DEVPOOL_OWNER_NAME, DEVPOOL_REPO_NAME, DEVPOOL_RFC_OWNER_NAME and DEVPOOL_RFC_REPO_NAME to .env (copy the values from .env.example)

xno-miner committed:
don't tweet rfcs

xno-miner committed:
don't tweet rfcs

xno-miner committed:
Moved the DEVPOOL_ variables back from env

xno-miner committed:
Improved code readability

xno-miner committed:
Minor fix

xno-miner committed:
Further improved readability

xno-miner committed:
Minor fix

xno-miner committed:
Create env variables for devpool/RFC

xno-miner committed:
Merge branch 'development' into development

xno-miner committed:
Create tests for RFCs

xno-miner committed:
Add missing files

xno-miner committed:
JSON.parse RFC env

xno-miner committed:
quotes in ymls

xno-miner committed:
Fix env and enable rfc test

xno-miner committed:
minor cleanup

xno-miner committed:
remove RFC env var, adjust tests

xno-miner commented:
The backend part is done (i think).

0x4007 commented on helpers/github.ts:
```diff
@@ -129,12 +130,24 @@ export function getDevpoolIssueLabels(issue: GitHubIssue, projectUrl: string) {
-    `Partner: ${ownerName}/${repoName}`, // partner
-    `id: ${issue.node_id}`, // id
-  ];
+  if (pricing != "Pricing: not set") {
```
Perhaps this should be an environment variable instead of a string

0x4007 replied:
I wonder if its a better idea to do `Pricing: -1` the idea is that I believe the parsers we use expect a number. We can just check if its less than 0. 

What is the best choice here @rndquu?

rndquu replied:
> I wonder if its a better idea to do `Pricing: -1` the idea is that I believe the parsers we use expect a number. We can just check if its less than 0.
> 
> What is the best choice here @rndquu?

The `Pricing: not set` is used purely to distinguish issues with and without price labels. The text itself is not used anywhere so it doesn't matter much whether it's `Pricing: -1`, `Pricing: not set`, etc...

@xno-miner Could you move the `Pricing: not set` string to a separate constant variable since it's used in multiple places ([one](https://github.com/xno-miner/devpool-directory/blob/8cdeed44ab0a42a694aca9d7641593e8dec66131/helpers/github.ts#L138), [two](https://github.com/xno-miner/devpool-directory/blob/8cdeed44ab0a42a694aca9d7641593e8dec66131/helpers/github.ts#L209))?

Otherwise the PR works fine.

xno-miner replied:
Sorry for the delay, I only just now have noticed this message. I pushed the changes.

0x4007 reviewed (commented):
Mixed feelings on my input because seems that there are many labels to keep track of

0x4007 commented:
> The backend part is done (i think).

Best that you run it on your own org/repos to test. Then link here as quality assurance.

xno-miner commented:
Good news: the bot syncs issues without price tags
Bad news: i accidentally flooded the official devpool-directory with RFCs from my "testbot6"

xno-miner committed:
Prevent from flooding the official devpool ever again

xno-miner committed:
maybe like this

xno-miner committed:
...

xno-miner committed:
?

xno-miner committed:
ok screw it

testbot6[bot] committed:
Update total rewards

xno-miner commented:
I think the real bot will close those issues automatically since they don't have price labels

Edit: It didn't. I also tried to make a modified version of the backend that will close all the issues but I can't modify issues. Unfortunately they have to be closed/removed manually by someone with permissions. Sorry for the inconvenience.

xno-miner commented:
> Best that you run it on your own org/repos to test. Then link here as quality assurance.

In the second attempt, after obliterating the issues tab on the official repo, i managed to run the bot in my own repo. I was worried for a moment that the labels are not working, but it seems like they were only empty in the ubiquity repo (lack of permissions maybe), but in my repo they are all there.

Here's the link: https://github.com/xno-miner/devpool-directory/issues

0x4007 commented:
@rndquu I think continuous deployment is running off of unmerged pulls 

I know at one point we had this problem in the past. Perhaps the CD code is outdated. 

https://github.com/ubiquity/devpool-directory-bounties/issues/30

rndquu commented:
@xno-miner 

1. This PR is ready for review, right?
2. How exactly did the `testbot6` flooded the https://github.com/ubiquity/devpool-directory repository? I mean how is that possible if `testbot6` is not even installed in the https://github.com/ubiquity/devpool-directory repository? What access token was used?

P.S. I'll remove all issues created by `testbot6` in https://github.com/ubiquity/devpool-directory/issues

Update: it seems that anybody is able to open new github issues unless [interaction limits](https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-repository#about-temporary-interaction-limits) are not set. The thing is that if we limit interactions to contributors or collaborators then new users won't be able to open PRs so this solution is not really handy for us. Created https://github.com/ubiquity/devpool-directory-bounties/issues/31.

xno-miner commented:
> This PR is ready for review, right?

Yes.

Keyrxng commented:
@xno-miner a couple of your comments made me laugh. These things happen don't stress, you've highlighted a problem which is a good thing. Good day ser 


EDIT: I only just noticed the commit titles lmaoooo 🤣

![image](https://github.com/user-attachments/assets/17642945-d2cf-4891-9d38-a017cfd463f2)

Reminds me of myself [here](https://github.com/ubiquibot/permit-generation/commit/881f006ca7f8dc64ba9fdeb7f02b3da8f0d68476) lol. But we tend to use conventional commits across the org for future reference dude

Keyrxng commented:
> > This PR is ready for review, right?
> 
> Yes.

@xno-miner if the PR is ready for review please convert this from a draft PR and request reviewers to look things over. Could you do this on all of your PRs that you need reviewed I think you have two or three drafts open, thanks.

xno-miner commented:
> > > This PR is ready for review, right?
> > 
> > 
> > Yes.
> 
> @xno-miner if the PR is ready for review please convert this from a draft PR and request reviewers to look things over. Could you do this on all of your PRs that you need reviewed I think you have two or three drafts open, thanks.

Sorry, I forgot they were drafts. I undrafted both of them.

0x4007 commented on package.json:
```diff
@@ -64,5 +64,6 @@
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
This should be removed.

xno-miner replied:
Sorry, for some reason it gets automatically added when I run it locally.

0x4007 replied:
Pretty sure it's `corepack` doing this @Keyrxng rfc

Keyrxng replied:
Correct it is indeed corepack

xno-miner committed:
fix: remove the line in package.json

rndquu commented:
@xno-miner 

On `yarn test` I'm getting the following errors locally:
```
 FAIL  tests/devpool-issue-handler.test.ts
  ● Test suite failed to run

    tests/devpool-issue-handler.test.ts:1495:50 - error TS2345: Argument of type '{ in: string[]; out: string[]; }' is not assignable to parameter of type '{ in: string[]; out: never[]; }'.
      Types of property '"out"' are incompatible.
        Type 'string[]' is not assignable to type 'never[]'.
          Type 'string' is not assignable to type 'never'.

    1495     const urls = Array.from(await getProjectUrls(opt));
                                                          ~~~

 FAIL  tests/twitter.test.ts
  ● Test suite failed to run

    helpers/github.ts:261:27 - error TS2339: Property 'split' does not exist on type 'never'.

    261     const len = orgOrRepo.split("/").length;
                                  ~~~~~

 FAIL  tests/helpers.test.ts
  ● Test suite failed to run

    helpers/github.ts:261:27 - error TS2339: Property 'split' does not exist on type 'never'.

    261     const len = orgOrRepo.split("/").length;
                                  ~~~~~

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |                   
----------|---------|----------|---------|---------|-------------------
Test Suites: 3 failed, 3 total
Tests:       0 total
Snapshots:   0 total
Time:        2.907 s
Ran all test suites.

```

My local ts environment also complains about `Property 'split' does not exist on type 'never'.`. The jest CI somehow stopped running for this PR. Can you reproduce those errors?

xno-miner commented:
> @xno-miner
> 
> On `yarn test` I'm getting the following errors locally:
> 
> ```
>  FAIL  tests/devpool-issue-handler.test.ts
>   ● Test suite failed to run
> 
>     tests/devpool-issue-handler.test.ts:1495:50 - error TS2345: Argument of type '{ in: string[]; out: string[]; }' is not assignable to parameter of type '{ in: string[]; out: never[]; }'.
>       Types of property '"out"' are incompatible.
>         Type 'string[]' is not assignable to type 'never[]'.
>           Type 'string' is not assignable to type 'never'.
> 
>     1495     const urls = Array.from(await getProjectUrls(opt));
>                                                           ~~~
> 
>  FAIL  tests/twitter.test.ts
>   ● Test suite failed to run
> 
>     helpers/github.ts:261:27 - error TS2339: Property 'split' does not exist on type 'never'.
> 
>     261     const len = orgOrRepo.split("/").length;
>                                   ~~~~~
> 
>  FAIL  tests/helpers.test.ts
>   ● Test suite failed to run
> 
>     helpers/github.ts:261:27 - error TS2339: Property 'split' does not exist on type 'never'.
> 
>     261     const len = orgOrRepo.split("/").length;
>                                   ~~~~~
> 
> ----------|---------|----------|---------|---------|-------------------
> File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
> ----------|---------|----------|---------|---------|-------------------
> All files |       0 |        0 |       0 |       0 |                   
> ----------|---------|----------|---------|---------|-------------------
> Test Suites: 3 failed, 3 total
> Tests:       0 total
> Snapshots:   0 total
> Time:        2.907 s
> Ran all test suites.
> ```
> 
> My local ts environment also complains about `Property 'split' does not exist on type 'never'.`. The jest CI somehow stopped running for this PR. Can you reproduce those errors?

I'm unable to reproduce the errors. Everything works correctly for me.

0x4007 commented:
The only place it matters is if tests pass in CI

xno-miner committed:
feat: Move "Pricing: not set" to const

0x4007 committed:
Merge branch 'development' into development-2

0x4007 reviewed (commented):
Testing this now at https://e93b0680.devpool-directory-ui.pages.dev/

Context https://github.com/ubiquity/work.ubq.fi/pull/77#issuecomment-2375328680
